lib/array_intersection.rb

```diff
@@ -1,6 +1,27 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m) because there is a nested loop, having to go through each element in both arrays
+# Space complexity: O(n) where n is the number of elements in the intersection array
 def intersection(array1, array2)
-  raise NotImplementedError
-end
+  intersection = []
+  if array1 == nil || array1.length == 0
+    return []
+  elsif array2 == nil || array2.length == 0
+    return []
+  else  
+    if array1.length > array2.length
+      array = array2
+      other_array = array1
+    else array = array1
+      other_array = array2
+    end
+      other_array.each do |num|
```

Comment:
If you are going to loop through each array with nested loops, it doesn't matter which one is smaller.